website/server/libs/cron.js

```diff
@@ -314,6 +314,13 @@ export function cron (options = {}) {
     let scheduleMisses = daysMissed;
 
     if (completed) {
+      if (task.history && task.history.length) {
+        const lastHistoryEntry = task.history[task.history.length - 1];
+        task.lastCompleted = moment(lastHistoryEntry.date).toDate();
```

Comment:
why do we need a new field instead of using the value from `task.history[task.history.length - 1] `?

Reply:
So a field like `task.lastCompleted` that's updated at the time that the Daily is marked completed by the user?
I did consider doing that but I realised that if the user completes a Daily by mistake then later un-completes it, we'd need to set `lastCompleted` back to its earlier value, which means we would have needed to store the earlier value in yet another field when we updated `lastCompleted` or we'd need to read the earlier value from the history array, and it seemed simpler to just use the history array in cron.

Reply:
sorry, ignore that, I think I misread your questions. a real answer will be here soon. :)

Reply:
The `task.lastCompleted` code was added by the original contributor and approved in https://github.com/HabitRPG/habitica/pull/8962/files but I can rework it to use the history array directly. It shouldn't be hard. I'll try to get that done in the next couple of days while it's fresh in my head!

Mostly as a reminder to myself:
Once that's approved and added back to delta, I'll remove the `lastCompleted` fields from the Dailies belonging to anyone who's used the delta test site.

Reply:
Ah sorry I missed that it wasn't added here! i'd say if it's not yet used and it's possible to use directly the history array it'd be better to do that

Reply:
It turns out to be not possible to use the history array directly, alas. Cron adds a history entry to each Daily that wasn't completed on the previous day, so the most recent entry isn't always the date that the Daily was last completed.

Unrelated to that, I'm adding more tests. I'll post to the PR when it's ready to be reviewed again.